modules/02_welfare/ineqLognormal/equations.gms

```diff
@@ -295,21 +296,21 @@ q02_budget_second(ttot,regi)$(ttot.val ge cm_startyear)..
 ***---------------------------------------------------------------------------
 $IFTHEN.INCONV %cm_INCONV_PENALTY% == "on"
 q02_inconvPen(t,regi)$(t.val > 2005)..
-    v02_inconvPen(t,regi)
+  v02_inconvPen(t,regi)
   =g=
-*' local air pollution for all entySe production except for coal solids (=sesofos), which is treated separately (see below)
-    SUM(pe2se(enty,entySe,te)$(NOT sameas(entySe,"sesofos")),
-        p02_inconvpen_lap(t,regi,te) * (vm_prodSe(t,regi,enty,entySe,te))
-    )
+*' local air pollution / inconvenience for all entySe production except for coaltr and biotrmod solids, wich are treated separately (see below)
+  SUM(pe2se(enty,entySe,te)$( NOT (sameas(te,"coaltr") OR sameas(te,"biotrmod") ) ),
+    p02_inconvpen_lap(t,regi,te) * vm_prodSe(t,regi,enty,entySe,te)
+  )
 ;
 
-q02_inconvPenCoalSolids(t,regi)$(t.val > 2005)..
-    v02_inconvPenCoalSolids(t,regi)
+q02_inconvPenSolidsBuild(t,regi)$(t.val > 2005)..
+  v02_inconvPenSolidsBuild(t,regi)
   =g=
-*' local air pollution for coal: inconvinienve penalty applies only for buildings use; slack variable ensures that v02_inconvPen can stay > 0
-    p02_inconvpen_lap(t,regi,"coaltr") * (vm_prodSe(t,regi,"pecoal","sesofos","coaltr")
-  - vm_cesIO(t,regi,"fesoi"))
-  + v02_sesoInconvPenSlack(t,regi)
+*' Local air pollution and inconvenience of using coal and (modern) biomass: inconvenience penalty applies only for use in residential/buildings
+*' The inconvenience of using traditional biomass are accounted for in v02_inconvPen, and thus additional to the penalty on using solids in residential
+  p02_inconvpen_lap(t,regi,"coaltr") * vm_demFeSector(t,regi,"sesofos","fesos","build","ES")
+  + p02_inconvpen_lap(t,regi,"biotrmod") * vm_demFeSector(t,regi,"sesobio","fesos","build","ES")
```

Comment:
I'd find it easier to read if you used the same structure as in the equation above so one sees the complement more clearly.
```suggestion
  SUM(pe2se(enty,entySe,te)$(sameas(te,"coaltr") OR sameas(te,"biotrmod") ),
    p02_inconvpen_lap(t,regi,te) * vm_demFeSector(t,regi,entySe,"fesos","build","ES")
  )
```

Reply:
I wanted explicitly to write it differently to highlight that this is NOT as above, where a technology gets multiplied by its output, but rather a specific multiplication of a technology parameter by a partially related other variable. So it is NOT something that could automatically be extended using an existing mapping or so. 

so I would keep it the way it currently is, given that you formulated "or ignored just as well" :-)